src/java/org/apache/cassandra/service/accord/txn/TxnWrite.java

```diff
@@ -86,23 +88,26 @@ public class TxnWrite extends AbstractKeySorted<TxnWrite.Update> implements Writ
     public static final TxnWrite EMPTY_CONDITION_FAILED = new TxnWrite(TableMetadatas.none(), Collections.emptyList(), false);
 
     private static final long EMPTY_SIZE = ObjectSizes.measure(EMPTY_CONDITION_FAILED);
+    private static final int FLAG_USER_TIMESTAMP = 0x01;
 
     public static class Update extends AbstractParameterisedVersionedSerialized<PartitionUpdate, TableMetadatas>
     {
-        private static final long EMPTY_SIZE = ObjectSizes.measure(new Update(null, 0, ByteBufferUtil.EMPTY_BYTE_BUFFER));
+        private static final long EMPTY_SIZE = ObjectSizes.measure(new Update(null, 0, ByteBufferUtil.EMPTY_BYTE_BUFFER, false));
         public final PartitionKey key;
         public final int index;
+        public final boolean userTimestamp;
 
-        public Update(PartitionKey key, int index, PartitionUpdate update, TableMetadatas tables)
+        public Update(PartitionKey key, int index, PartitionUpdate update, TableMetadatas tables, boolean userTimestamp)
```

Comment:
I don't think this is the right granularity for `userTimestamp` because it's a whole partition update which is multiple rows some of which might preserve timestamp and some of which might not.

We already capture what to do with timestamps with `preserveTimestamps` for the whole partition update so it's redundant to add `userTimestamp`?